server/src/main/java/io/spine/server/aggregate/Aggregate.java

```diff
@@ -392,19 +423,125 @@ protected String getMissingTxMessage() {
 
     /**
      * Transforms the current state of the aggregate into the {@link Snapshot} instance.
+     * 
+     * <p> The aggregate is aware of the snapshot being made. For example, this information is used 
+     * by {@link #didHandleSinceLastSnapshot}.
      *
      * @return new snapshot
      */
     @CheckReturnValue
-    Snapshot toSnapshot() {
+    Snapshot makeSnapshot() {
         final Any state = AnyPacker.pack(getState());
         final Snapshot.Builder builder = Snapshot.newBuilder()
                 .setState(state)
                 .setVersion(getVersion())
                 .setTimestamp(getCurrentTime());
+        final Snapshot snapshot = builder.build();
+        historySinceLastSnapshot.clear();
+        return snapshot;
+    }
+
+    /**
+     * Creates an iterator of the aggregate event history with reverse traversal.
+     *
+     * <p>The records are returned sorted by timestamp in a descending order (from newer to older).
+     * 
+     * <p>The iterator is empty if there's no history for the aggregate.
+     *
+     * @param  criteria specifies the filters the returned data should pass and how the query 
+     *                  should be performed
+     * @return new iterator instance
+     */
+    public Iterator<AggregateEventRecord> historyBackward(AggregateRecordQueryCriteria criteria) {
+        checkNotNull(criteria);
+        final int batchSize = criteria.batchSize();
+        final List<AggregateEventRecord> items = queryBatch(batchSize);
+        return items.iterator();
+    }
+
+    /**
+     * Obtains a batch of aggregate records specified by the batch size.
+     * 
+     * @param batchSize the amount of events that must be retrieved
+     */
+    private List<AggregateEventRecord> queryBatch(int batchSize) {
```

Comment:
What if the batch size is greater than the recent history depth?

That's another point to simplify the `historyBackward` public API. Let's not allow to set the size. At least for now, while we aren't able to load additional (older, than those "recent") events from the `EventStore`.